Projects/Features/PlaylistFeature/Sources/ViewControllers/PlayListDetailViewController.swift

```diff
@@ -326,123 +174,186 @@ extension PlayListDetailViewController {
-        }
-        .disposed(by: disposeBag)
-
-        tableView.rx.itemSelected
-            .withLatestFrom(output.dataSource) { ($0, $1) }
-            .withLatestFrom(input.state) { ($0.0, $0.1, $1) }
-            .filter { $0.2.isEditing == false }
-            .subscribe(onNext: { [weak self] indexPath, dataSource, _ in
-                guard let self else { return }
+        currentState.map(\.selectedItemCount)
+            .subscribe(onNext: { [weak self] count in
 
-                guard let type = self.viewModel.type else {
+                guard let self, let type = self.reactor?.type else {
                     return
                 }
 
                 switch type {
                 case .custom:
-                    let song: SongEntity = dataSource[indexPath.section].items[indexPath.row]
-                    playState.loadAndAppendSongsToPlaylist([song])
+                    break
                 case .wmRecommend:
-                    input.songTapped.onNext(indexPath.row)
+                    if count == 0 {
+                        self.hideSongCart()
+                    } else {
+                        self.showSongCart(
+                            in: self.view,
+                            type: .WMPlayList,
+                            selectedSongCount: count,
+                            totalSongCount: self.reactor?.currentState.dataSource.first?.items.count ?? 0,
+                            useBottomSpace: false
+                        )
+                        self.songCartView?.delegate = self
+                    }
                 }
 
             })
             .disposed(by: disposeBag)
 
-        output.groupPlaySongs
-            .subscribe(onNext: { [weak self] songs in
-                guard let self = self else { return }
-                self.playState.loadAndAppendSongsToPlaylist(songs)
+        currentState.map(\.isEditing)
+            .subscribe(onNext: { [weak self] flag in
+
+                guard let self else { return }
+
+                self.navigationController?.interactivePopGestureRecognizer?.delegate = flag ? self : nil
+                self.moreButton.isHidden = flag
+                self.editPlayListNameButton.isHidden = !flag
+                self.editStateLabel.isHidden = !flag
+                self.completeButton.isHidden = !flag
+
+                tableView.isEditing = flag
+                tableView.reloadData()
+
             })
             .disposed(by: disposeBag)
+    }
+
+    override public func bindAction(reactor: PlaylistDetailReactor) {
+        super.bindAction(reactor: reactor)
 
-        Utility.PreferenceManager.$startPage
-            .skip(1)
+        reactor.action.onNext(.viewDidLoad)
+
+        tableView.rx.itemSelected
+            .filter { _ in reactor.type == .wmRecommend }
+            .map { $0.row }
+            .subscribe(onNext: {
+                reactor.action.onNext(.tapSong($0))
+            })
+            .disposed(by: disposeBag)
```

Comment:
```suggestion
        tableView.rx.itemSelected
            .filter { _ in reactor.type == .wmRecommend }
            .map { $0.row }
            .map(Reactor.Action.tapSong)
            .bind(to: reactor.action)
            .disposed(by: disposeBag)
```
이런식으로 하는거 어떤가요?

Reply:
```suggestion
        tableView.rx.itemSelected
            .filter { _ in reactor.type == .wmRecommend }
            .map { $0.row }
            .map { Reactor.Action.tapSong($0) }
            .bind(to: reactor.action)
            .disposed(by: disposeBag)
```
생략이 별로인거같으면 이렇게 해도 될거같네용

Reply:
적용